src/vs/base/test/common/encoding/encoding.test.ts

```diff
@@ -4,10 +4,13 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as assert from 'assert';
+// FIX BEFORE MERGE
+// eslint-disable-next-line code-import-patterns
```

Comment:
From my understanding it makes sense for tests to live next to their respective modules. It makes the structure a little bit more straightforward. That's why I move `encoding.test.ts` to the `common` directory.

Here these tests are using `fs` which violates layers rule. However `fs` is used purely to read fixtures, so in my opinion it falls under the same exception as `assert` for example.

I see 2 options for `fs`:
1. Add it to list of exceptions for the rule - this will have a side effect of allowing it everywhere in common tests
2. Just keep `eslint-disable-next-line` here with the comment of why it is ignored

Which way do you think is better?

Reply:
@gyzerok we cannot have `fs` in `common` though because those tests would not run in our browser unit environment. We shim the `assert` module, so that one is ok. Please move this file back to the `node` layer.